src/Factory/TermFactory.php

```diff
@@ -60,9 +60,16 @@ protected function from_id(int $id): ?Term
         return $this->build($wp_term);
     }
 
-    protected function from_wp_term_query(WP_Term_Query $query): iterable
+    protected function from_wp_term_query(WP_Term_Query $query) 
     {
-        return \array_map([$this, 'build'], $query->get_terms());
+        $terms = $query->get_terms();
+        $_fields = $query->query_vars['fields'];
+        if ('all' === $_fields || 'all_with_object_id' === $_fields) {
+            return \array_map([$this, 'build'], $terms);
+        }
+        else {
```

Comment:
We don’t need an else statement here but could directly return `$terms`.

Reply:
Done